plugins/modules/aci_l3out.py

```diff
@@ -307,26 +386,84 @@ def main():
     enforceRtctrl = module.params.get("route_control")
     vrf = module.params.get("vrf")
     l3protocol = module.params.get("l3protocol")
+    ospf = module.params.get("ospf")
     asn = module.params.get("asn")
     state = module.params.get("state")
     tenant = module.params.get("tenant")
     name_alias = module.params.get("name_alias")
 
     if l3protocol:
         if "eigrp" in l3protocol and asn is None:
-            module.fail_json(msg="Parameter 'asn' is required when l3protocol is 'eigrp'")
+            module.fail_json(
+                msg="Parameter 'asn' is required when l3protocol is 'eigrp'"
+            )
         if "eigrp" not in l3protocol and asn is not None:
-            module.warn("Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored")
+            module.warn(
+                "Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored"
+            )
 
     enforce_ctrl = ""
     if enforceRtctrl is not None:
         if len(enforceRtctrl) == 1 and enforceRtctrl[0] == "import":
-            aci.fail_json("The route_control parameter is invalid: allowed options are export or import,export only")
+            aci.fail_json(
+                "The route_control parameter is invalid: allowed options are export or import,export only"
+            )
         elif len(enforceRtctrl) == 1 and enforceRtctrl[0] == "export":
             enforce_ctrl = "export"
         else:
             enforce_ctrl = "export,import"
-    child_classes = ["l3extRsL3DomAtt", "l3extRsEctx", "bgpExtP", "ospfExtP", "eigrpExtP", "pimExtP"]
+    else:
+        enforce_ctrl = "export"
+
+    child_classes = [
+        "l3extRsL3DomAtt",
+        "l3extRsEctx",
+        "bgpExtP",
+        "ospfExtP",
+        "eigrpExtP",
+        "pimExtP",
+    ]
+
+    child_configs = [
+        dict(l3extRsL3DomAtt=dict(attributes=dict(tDn="uni/l3dom-{0}".format(domain)))),
+        dict(l3extRsEctx=dict(attributes=dict(tnFvCtxName=vrf))),
+    ]
+    if l3protocol is not None:
+        l3protocol_child_configs = dict(
+            bgp=dict(bgpExtP=dict(attributes=dict(status="deleted"))),
+            eigrp=dict(eigrpExtP=dict(attributes=dict(status="deleted"))),
+            ospf=dict(ospfExtP=dict(attributes=dict(status="deleted"))),
+            pim=dict(pimExtP=dict(attributes=dict(status="deleted"))),
+        )
+        for protocol in l3protocol:
+            if protocol == "bgp":
+                l3protocol_child_configs["bgp"] = dict(
+                    bgpExtP=dict(attributes=dict(descr="", nameAlias=""))
```

Comment:
Do we need to pass in empty strings when we are not providing user way to manipulate these attributes or would an empty dict be enough? Same applies for the other protocols.

Reply:
an empty dict would be enough.